### PR TITLE
adds support for incremental output of completed tasks in the simulator

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- adds support for incremental output of completed tasks in the simulator

## Why are we making these changes?

We would like to get quicker feedback than having to wait for the simulator to end running before seeing the list of completed tasks.

### CODE REVIEW

File: scheduler/test/cook/test/simulator.clj [hunk 1]
```diff
@@ -325,12 +340,42 @@
...
                                                  :lingering-task-trigger-chan (async/chan)
                                                  :cancelled-task-trigger-chan (async/chan)}})
-        {:keys [time-ms-between-optimizer-calls time-ms-between-rebalancing]} config]
+        {:keys [time-ms-between-incremental-output time-ms-between-optimizer-calls time-ms-between-rebalancing]} config]
+    ;; launch a thread to perform incremental writes of completed jobs
+    (when temp-out-trace-file
+      (let [last-start-time-atom (atom simulation-time)
+            write-thread (Thread.
+                           ^Runnable
+                           (fn write-completed-tasks []
```
Comment (dposada) [inline scheduler/test/cook/test/simulator.clj]: It might be nice to extract this to a `defn`

File: scheduler/test/cook/test/simulator.clj [hunk 2]
```diff
@@ -325,12 +340,42 @@
...
+                                     (do
+                                       (log/info "Writing" (count completed-task-ents) "tasks to temporary output for simulation time" current-time)
+                                       (dump-jobs-to-csv completed-task-ents temp-out-trace-file
+                                                         :add-headers (= start-time simulation-time)
+                                                         :append true))
+                                     (log/info "No completed tasks in last" time-ms-between-incremental-output "ms"))
+                                   (reset! last-start-time-atom current-time))
+                                 (catch Throwable ex
+                                   (log/error ex "Error writing incremental completed job output!"))))))]
+        (log/info "temporary output will be updated every" time-ms-between-incremental-output "ms")
```
Comment (dposada) [inline scheduler/test/cook/test/simulator.clj]: The other log lines start with a capitalized word, but this one doesn't

Comment (shamsimam) [reply scheduler/test/cook/test/simulator.clj]: Changed to `Temporary`

File: scheduler/test/cook/test/simulator.clj (hunk 1 above)
Comment (shamsimam) [reply scheduler/test/cook/test/simulator.clj]: Extracted helper function `write-completed-tasks`. I did not extract the whole function as I did not want to expose the `last-start-time-atom` atom.

**Pull request merged**